documentation/contributing/maintainer_pr/index.markdown

```diff
@@ -5,65 +5,84 @@ slug: maintainer_pr
+
+However, for larger changes to the code, it's absolutely desired to organize the pull request into multiple, clearly separated, but interrelated commits. In such cases, it is usually also meaningful to keep this commit history, thus being able to track these individual commits also later in the future. In this case, a maintainer should ask the contributor to cleanup "fixup" commits via squashing or interactive rebasing before eventually performing a proper `merge commit`.
 
-### The Latest Branch Preserves All Original Commits
+Multiple, independent commits should not show up in a single PR, but should be split into multiple independent PRs. GitHub's `rebase-merge` policy, which could be used in such cases, unfortunately drops any notion of the related pull request id. To summarize:
+
+- Single commits -> squash
+- Multiple fixup-style commits -> squash
+- Multiple **clean**, interrelated commits -> merge commit
+- Multiple **clean**, independent commits -> rebase and merge
+
+Contributors should indicate their desired merge-policy in the main PR comment.
+
+## The Latest Branch Preserves All Original Commits
 
 All contributions that get merged into MoveIt and should be included in later ROS distributions have to end up in the latest branch with their full description and individual commits.
 This is necessary in order to preserve the history of the changes.
 Future versions of MoveIt might not share the whole history of older branches and finding your `git blame` returns a commit called ``cherry-picked #123``, without an additional description, is not desirable.
 
-### Merge Only If You Make Sure It Becomes Available In Later Branches Too
+## Merge Only If You Make Sure It Becomes Available In Later Branches Too
 
 Don't merge a request unless you make sure the same (or a similar) patch-set is merged into all later branches at the same time.
 We don't want to lose changes that are available in one ROS distribution because someone forgot to do that.
 
+Use mergify bot to automatically ensure changes get cherry picked to other appropriate branches.
+
 If the commits apply cleanly to future branches and there is no evidence that they will break anything there, you are free to add the changes to the respective branches.
 (Make sure you respect the all-commits-in-latest-branch guideline though).
 
 Otherwise either the requestor or the maintainer should create new pull-requests targeting the later branches.
 If possible, merge these *together* with the original request.
 In this context it might be worth spending some time on making use of features available in later ROS distributions to simplify the code, e.g. by using a new coding standard or a more current version of a library.
```

Comment:
I don't like the practice of filing several identical PRs for several branches in parallel. This increases the review effort and introduces the risk of several uncoordinated reviews. As stated above, a change should be first targeted at and accepted for the main branch and only subsequently get backported.

Reply:
I did not change or write this section and am not sure what you want instead. Please open new PR for this.